textdb/textdb-textql/src/main/javacc/TextQLParser.jj

```diff
@@ -0,0 +1,697 @@
+    // return generated Statement
+    {
+        return newStatement;
+    }
+}
+
+/**
+ * Consume a "CREATE VIEW" statement declaration form the input and generate a
+ * CreateViewStatement object with the parsed data
+ * Examples of valid input:
+ *         CREATE VIEW viewname0 AS <SelectStatement>
+ * Grammar: "CREATE" "VIEW" <Identifier>
+ *          "AS" ( <SelectStatement> )
+ * @return The CreateViewStatement representation of the statement declaration
+ */
+CreateViewStatement createViewStatement() : 
+{
+    CreateViewStatement createViewStatement = new CreateViewStatement();
+    Statement innerStatement;
+    String viewName;
+}
+{
+    // parse CREATE VIEW viewName AS
+    <CREATE>
+    <VIEW>
+    viewName = identifierLiteralToString() { createViewStatement.id = viewName; }
+    <AS>
+    // parse the inner statement (subStatement) as a SelectStatement
+    (
+        innerStatement = selectStatement()
+    ){ createViewStatement.innerStatement = innerStatement; }
+    // return generated CreateViewStatement
+    {
+        return createViewStatement;
+    }
+}
+
+/**
+ * Parse a select statement declaration form the input and generate a
+ * SelectStatement object with the parsed data
+ * Examples of valid input:
+ *         SELECT * FROM t
+ *         SELECT a, b, c FROM t LIMIT 1 OFFSET 8
+ *         EXTRACT KEYWORDMATCH(g0, "key1") FROM k
+ *         SELECT a EXTRACT KEYWORDMATCH(field, "key1") FROM k
+ *         SELECT * EXTRACT KEYWORDMATCH([h6,h7,k8,k9], "key1", conjunction) FROM k LIMIT 1 OFFSET 8
+ * Grammar: (
+ *              "SELECT" ( "*" | <identifierListToListString> ) ( "EXTRACT" <ExtractPredicate> )?
+ *            |
+ *              "EXTRACT" <ExtractPredicate>
+ *          )
+ *          "FROM" <Identifier>
+ *          ("LIMIT" <Number> )?
+ *          ("OFFSET" <Number> )?
+ * @return The SelectStatement representation of the statement declaration
+ */  
+SelectStatement selectStatement() : 
+{
+    SelectStatement selectStatement = new SelectStatement();
+    selectStatement.id = getNewStatementId();
+    List<String> identifiers;
+    String fromClause;
+    Integer limitClause;
+    Integer offsetClause;
+    ExtractPredicate extractPredicate;
+}
+{
+    // parse the SELECT clause or/and the EXTRACT clause
+    (
+            // parse only the SELECT clause or parse the SELECT clause and the EXTRACT clause
+            (
+                // parse the SELECT clause
+                <SELECT> 
+                ( 
+                        <STAR> { selectStatement.projectAll = true; } // SELECT *
+                    | 
+                        identifiers = identifierListToListString() { selectStatement.projectedFields = identifiers; } // SELECT a,b,...
+                )
+                // parse the EXTRACT clause (optional)
+                (
+                    <EXTRACT> extractPredicate = extractPredicate() { selectStatement.extractPredicate = extractPredicate; } // EXTRACT extractPredicate
```

Comment:
This line is too long.  In general, we want each line to be at most 80 characters.  Can you check globally?